packages/plugin-metrics/test/unit/spec/batcher.js

```diff
@@ -0,0 +1,168 @@
+/**!
+ *
+ * Copyright (c) 2015 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {assert} from '@ciscospark/test-helper-chai';
+import lolex from 'lolex';
+import Metrics, {config} from '../..';
+import MockSpark from '@ciscospark/test-helper-mock-spark';
+import sinon from '@ciscospark/test-helper-sinon';
+import {SparkHttpError} from '@ciscospark/spark-core';
+
+function promiseTick(count) {
+  let promise = Promise.resolve();
+  while (count > 1) {
+    promise = promise.then(() => {
+      return promiseTick(1);
```

Comment:
if i'm understanding your question correctly, that would produce a very short synchronous wait. the goal of promiseTick is to let `count` ticks pass by chaining `count` `promise.then`s.

basically, it's a hack to deal with the fact that lolex replaces the clock used by setTimeout and setInterval, but not the clock used by promises.